packages/providers/src.ts/base-provider.ts

```diff
@@ -1256,21 +1256,32 @@ export class BaseProvider extends Provider implements EnsProvider {
         }
     }
 
-    async getCode(addressOrName: string | Promise<string>, blockTag?: BlockTag | Promise<BlockTag>): Promise<string> {
+    /**
+     *  Get contract bytecode implementation, using the REST Api.
+     *  It returns the bytecode, or a default value as string.
+     *
+     * @param addressOrName The address to obtain the bytecode of
+     */
+    async getCode(addressOrName: string | Promise<string>, throwOnNonExisting?: boolean): Promise<string> {
         await this.getNetwork();
-        const params = await resolveProperties({
-            address: this._getAddress(addressOrName),
-            blockTag: this._getBlockTag(blockTag)
-        });
-
-        const result = await this.perform("getCode", params);
+        addressOrName = await addressOrName;
+        const { shard, realm, num } = getAccountFromAddress(addressOrName);
+        const shardNum = BigNumber.from(shard).toNumber();
+        const realmNum = BigNumber.from(realm).toNumber();
+        const accountNum = BigNumber.from(num).toNumber();
```

Comment:
nit: I think we should extract this as an utility function

Reply:
We're going to implement such function in a separate PR.